modules/experimental/src/categories/table/columnar-table-batch.js

```diff
@@ -25,10 +25,18 @@ export default class ColumnarTableBatch {
   }
 
   getNormalizedBatch() {
-    const columns = this.columns;
+    this.pruneColumns();
+    const columns = Array.isArray(this.schema) ? this.columns : {};
+
+    if (!Array.isArray(this.schema)) {
```

Comment:
What happens if schema is an array?

Is `schema` required or can it be null?

Reply:
`schema` is an array if there is no header row.